lib/common/components/FeedLabel.js

```diff
@@ -48,52 +48,78 @@ class FeedLabel extends React.Component<Props, State> {
     this.refs.deleteModal.open()
   }
 
+  _getWrapperClasses = () => {
+    const classes = ['feedLabelWrapper']
+    if (this._isEditable()) classes.push('withButtons')
+    return classes.join(' ')
+  }
+
+  _getLabelClasses = () => {
+    const classes = ['feedLabel']
+    classes.push('smaller')
+    if (this._isCheckable()) classes.push('clickable')
+    return classes.join(' ')
+  }
+
+  _isCheckable = () => this.props.checked !== undefined
+
   _onClickEdit = () => {
     this.refs.editModal.open()
   }
 
-  render () {
-    const {small, label, checked, onClick: labelOnClick, user} = this.props
+  _onClick = () => {
+    const {label, onClick} = this.props
+    onClick && onClick(label.id)
+  }
+
+  _isEditable = () => {
+    const {editable, label, user} = this.props
+    if (!editable) return false
     const projectAdmin =
       user &&
       user.permissions &&
       user.permissions.isProjectAdmin(label.projectId)
+    return projectAdmin
+  }
+
+  render () {
+    const {label, checked = false} = this.props
+
+    // Used to avoid collision when label is rendered multiple times
+    const uniqueId = `${label.id}-${Date.now().toString(36)}`
 
     return (
-      <div className='feedLabelWrapper'
-        onClick={!labelOnClick ? null : () => labelOnClick(label.id)}
-        onKeyPress={!labelOnClick ? null : () => labelOnClick(label.id)}
-        role='checkbox'
-        aria-checked={checked}
-        tabIndex={0}
-      >
-        {checked !== undefined && (
+      <div className={this._getWrapperClasses()}>
+        {this._isCheckable() && (
           <div className='actionButtons'>
-            <input type='checkbox' checked={checked} />
+            <input
+              type='checkbox'
+              checked={checked}
+              onClick={this._onClick}
+              onKeyPress={this._onClick}
+              aria-checked={checked}
+              tabIndex={0}
+              id={uniqueId}
+            />
           </div>
         )}
-        <div
-          className={`feedLabel ${small ? 'smaller' : ''} 
-        ${
-  // Need an explicit check for undefined, since it is expected
-  // to be true/false
-      checked !== undefined ? 'clickable' : ''
-      }`}
+        <label
```

Comment:
Why not, although for semantics, when there is no checkbox, the label refers to an element that does not exist (that's why I had it in a particular way in #697).

Reply:
Ah this is a good point. Could try to make the for field optional? Or could make the syntactic argument that it's technically still a label for a certain checkbox it's just that that checkbox doesn't exist haha

Reply:
In #697 I had two separate renderings one when a checkbox was displayed and one otherwise.

Reply:
Tried to do this the cool way and have optional props via spread, but our accessibility eslint rule required a for for a label. 

@landonreed do you think it's worth doing a second render option?

Reply:
Nope, I'm fine with it as is.